test/com/google/enterprise/adaptor/filenet/FileNetAdaptorTest.java

```diff
@@ -73,4 +97,282 @@ public void testGetConnection() throws Exception {
+  public void testGetDocIds() throws Exception {
+    long startTime = new Date().getTime();
+    RecordingDocIdPusher pusher
+        = (RecordingDocIdPusher) context.getDocIdPusher();
+    config.overrideKey("feed.maxUrls", "5");
+    adaptor.init(context);
+    adaptor.getDocIds(pusher);
+
+    List<Record> golden = ImmutableList.of(
+        new Record.Builder(new DocId("pseudo/type=document"))
+        .setCrawlImmediately(true).build());
+    assertEquals(golden, pusher.getRecords());
+  }
+
+  @Test
+  public void testGetDocContent_invalidDocId1() throws Exception {
+    adaptor.init(context);
+
+    List<String> messages = new ArrayList<String>();
+    captureLogMessages(FileNetAdaptor.class, "Invalid DocId:", messages);
+    RecordingResponse response = new RecordingResponse();
+    adaptor.getDocContent(new MockRequest(new DocId("foo")),
+        response);
+    assertEquals(RecordingResponse.State.NOT_FOUND, response.getState());
+    assertEquals(messages.toString(), 1, messages.size());
+  }
+
+  @Test
+  public void testGetDocContent_invalidDocId2() throws Exception {
+    adaptor.init(context);
+
+    List<String> messages = new ArrayList<String>();
+    captureLogMessages(FileNetAdaptor.class, "Invalid DocId:", messages);
+    RecordingResponse response = new RecordingResponse();
+    adaptor.getDocContent(new MockRequest(new DocId("foo/bar")),
+        response);
+    assertEquals(RecordingResponse.State.NOT_FOUND, response.getState());
+    assertEquals(messages.toString(), 1, messages.size());
+  }
+
+  @Test
+  public void testGetDocContent_invalidDocId3() throws Exception {
+    adaptor.init(context);
+
+    List<String> messages = new ArrayList<String>();
+    captureLogMessages(FileNetAdaptor.class, "Unsupported type:", messages);
+    RecordingResponse response = new RecordingResponse();
+    adaptor.getDocContent(new MockRequest(new DocId("pseudo/type=foo")),
+        response);
+    assertEquals(RecordingResponse.State.NOT_FOUND, response.getState());
+    assertEquals(messages.toString(), 1, messages.size());
+  }
+
+  @Test
+  public void testGetDocContent_shortCheckpoint() throws Exception {
+    config.overrideKey("feed.maxUrls", "5");
+    adaptor.init(context);
+
+    long startTime = new Date().getTime();
+    RecordingResponse response = new RecordingResponse();
+    adaptor.getDocContent(new MockRequest(new DocId("pseudo/type=document")),
+        response);
+
+    List<Record> golden = ImmutableList.of(
+        new Record.Builder(
+            new DocId("guid/{AAAAAAAA-0000-0000-0000-000000000001}"))
+        .build(),
+        new Record.Builder(
+            new DocId("guid/{AAAAAAAA-0000-0000-0000-000000000002}"))
+        .build(),
+        new Record.Builder(
+            new DocId("guid/{AAAAAAAA-0000-0000-0000-000000000003}"))
+        .build(),
+        new Record.Builder(
+            new DocId("guid/{AAAAAAAA-0000-0000-0000-000000000004}"))
+        .build());
+
+    RecordingDocIdPusher pusher
```

Comment:
The fact that's wired in another file isn't helping, but a comment on the helper method should be enough. Still tempted to wire it in from here, especially so that we can ensure fresh instances in `@Before` or call reset in `@After` (you can't depend on a single class instance per test method; it's a JUnit config option).